src/main/java/uk/gov/hmcts/reform/sscs/util/SscsUtil.java

```diff
@@ -131,6 +132,37 @@ private static void updateTranslationStatus(SscsCaseData caseData, SscsDocumentT
         }
     }
 
+    public static DocumentType getWriteFinalDecisionDocumentType(SscsCaseData caseData, boolean isPostHearingsEnabled) {
+        if (isPostHearingsEnabled
+            && isYes(caseData.getPostHearing().getCorrection().getCorrectionFinalDecisionInProgress())) {
+            return DocumentType.DRAFT_CORRECTED_NOTICE;
+        }
+
+        return DocumentType.DRAFT_DECISION_NOTICE;
+    }
+
+    public static DocumentType getIssueFinalDecisionDocumentType(SscsCaseData caseData, boolean isPostHearingsEnabled) {
+        if (isPostHearingsEnabled
+            && isYes(caseData.getPostHearing().getCorrection().getCorrectionFinalDecisionInProgress())) {
+            return DocumentType.CORRECTION_GRANTED;
+        }
+
+        return DocumentType.FINAL_DECISION_NOTICE;
+    }
+
+    public static void setCorrectionInProgress(CaseDetails<SscsCaseData> caseDetails, boolean isPostHearingsEnabled) {
+        if (isPostHearingsEnabled) {
+            Correction correction = caseDetails.getCaseData().getPostHearing().getCorrection();
+
+            if ((State.POST_HEARING.equals(caseDetails.getState())
+                    || State.DORMANT_APPEAL_STATE.equals(caseDetails.getState()))) {
+                correction.setCorrectionFinalDecisionInProgress(YesNo.YES);
+            } else {
+                correction.setCorrectionFinalDecisionInProgress(YesNo.NO);
+            }
```

Comment:
Extract the if into a YesNo called correctionInProgress. Then do correction.setCorrectionFinalDecisionInProgress(correctionInProgress)

Reply:
done